README.md

```diff
@@ -233,14 +233,19 @@ Finally prune all unreferenced old RPM files:
 bazeldnf prune --workspace /my/WORKSPACE --buildfile /my/BUILD.bazel
 ```
 
+By default `bazeldnf rpmtree` will try to find a solution which only contains
+the newest packages of all involved repositories. If pinning versions is
+important, the `--nobest` option can be supplied. With this option all packages
+are considered. Newest packages will have the higest weight but it may not
+always be able to choose them and older packages may be pulled in instead.
```

Comment:
I will happily admit that the whole dependency resolution thing goes way above my head, but why is specifying `--nobest` necessary when pinning packages? If I provide a specific versions to the resolver, shouldn't it be enough to make that version of the package weigh way above any other version to ensure that the resolver will pick the intended solution?

Reply:
> I will happily admit that the whole dependency resolution thing goes way above my head, but why is specifying `--nobest` necessary when pinning packages? If I provide a specific versions to the resolver, shouldn't it be enough to make that version of the package weigh way above any other version to ensure that the resolver will pick the intended solution?

The pinned versions will be taken into account with or without `--nobest`. The only issue is that pinned versions may require other not-newest packages. `bazeldnf` by default will only allow newest version or pinned versions. If you specify `--nobest` it will allow indirect not-newest dependencies too (still trying to not use them wherever possible). That may differ from `dnf --nobest`. I could not figure out entirely what `bazeldnf --nobest` actually does in detail so far.

Reply:
The explanation makes sense.

One small note: I'm not sure what the exact semantics of `dnf --nobest` are either, but ideally the identically-named options would have identical behavior to avoid confusion...

Reply:
> One small note: I'm not sure what the exact semantics of `dnf --nobest` are either, but ideally the identically-named options would have identical behavior to avoid confusion...

Yes, it works pretty similar already, but I have to check a few corner cases. That is definitely my goal.

Reply:
Updated the description to make it more clear.